sites/geohub/src/routes/(app)/management/stac/api/[id]/+layout.server.ts

```diff
@@ -6,17 +6,28 @@ import { generateHashKey } from '$lib/helper';
 
 export const load: LayoutServerLoad = async ({ params, fetch }) => {
 	const id = params.id;
+
 	const titilerUrl = env.TITILER_ENDPOINT?.replace('/cog', '') ?? '';
 
 	const stac = await getSTAC(id);
-	const datasetId = generateHashKey(stac.url);
-	const res = await fetch(`/api/datasets/${datasetId}`);
-	const isRegistered = res.status !== 404;
+
+	const collectionUrl = `${stac.url}collections/sentinel-2-l2a/items`;
+
+	const datasetId = generateHashKey(collectionUrl);
+
+	const datasetRes = await fetch(`/api/datasets/${datasetId}`);
+
+	const dataset = await datasetRes.json();
+
+	const isRegistered = datasetRes.status !== 404;
+
 	if (!stac) {
 		error(404, `This stac ID (${id}) is not found.`);
 	}
+
 	return {
 		stac,
+		dataset,
```

Comment:
I cannot understand why you need to return dataset  at +layout.server.ts in stac management pages.

if dataset is necessary for `tool`, you can fetch dataset inside StacApiExplorer component when users select Tools tab. I think that is waht current STAC api component is doing for assets and product.

So, you basically don't need to export  `dataset` prop from StacApiExplorer